apps/docs/components/ui/table/TypographyTable.tsx

```diff
@@ -38,6 +41,32 @@ function transformDataToTokenData(inputData: Record<string, { name: string; valu
 function groupItemsByPropertiesAndSizes(tokenData: TokenData, itemType: string): GroupedItems {
     const groupedItems: GroupedItems = {} as GroupedItems;
 
+    if (itemType === "overline") {
+        const overlineItem = {} as Record<typeof FontProperties[number], string>;
+
+        FontProperties.forEach(property => {
+            const propertyKey = property;
+
+            if (!tokenData[propertyKey]) {
+                return;
+            }
+
+            const matchingItem = tokenData[propertyKey].find(item => {
+                const nameParts = item.name.split("-");
+
+                return nameParts.includes(itemType);
+            });
+
+            if (matchingItem) {
+                overlineItem[propertyKey] = matchingItem.value;
+            }
+        });
+
+        groupedItems["overline"] = overlineItem;
+
+        return groupedItems;
+    }
+
     Sizes.forEach(size => {
```

Comment:
You should really try to simplify all the code in here. There is too many "if overline". The only "real" if overline should be to add the textTransformation uppercase. 

All the other ifs should be "if there is no sizes (this only match overline) , don't display the name column".

You can create a token structure like this: 
```
{
    "heading": {
        "xs" : {
            fontSize: {
                name: "hop-heading-xs-font-size",
                value: "1rem"
            },
            fontWeight .......
        },
       "sm" .........
    }, 
   "overline": {
            fontSize: {
                name: "hop-overline-font-size",
                value: "0.75rem"
            },
    }
}
```

this way, if one of the typo category doesn't have sizes in it, you can skip the name column